File: stasis-core/src/de/boxxit/stasis/serializer/AuthenticationMissmatchExceptionSerializer.java
package de.boxxit.stasis.serializer;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryo.Serializer;
import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;
import de.boxxit.stasis.AuthenticationMissmatchException;

/**
 * User: Christian Fruth
 */
public class AuthenticationMissmatchExceptionSerializer extends Serializer<AuthenticationMissmatchException>
{
	public AuthenticationMissmatchExceptionSerializer()
	{
	}

	@Override
	public void write(Kryo kryo, Output output, AuthenticationMissmatchException object)
	{
		output.writeBoolean(object.isAuthenticated());
	}

	@Override
	public AuthenticationMissmatchException read(Kryo kryo, Input input, Class<AuthenticationMissmatchException> type)
	{
		boolean authenticated = input.readBoolean();
		return new AuthenticationMissmatchException(authenticated);
	}
}

File: stasis-ios/Stasis/AuthenticationMissmatchExceptionSerializer.m
//
//  AuthenticationMissmatchExceptionSerializer.m
//  Stasis
//
//  Created by Christian Fruth on 06.03.13.
//  Copyright (c) 2013 Boxx IT Solutions e.K. All rights reserved.
//

#import "AuthenticationMissmatchExceptionSerializer.h"
#import "AuthenticationMissmatchException.h"
#import "Kryo.h"

@implementation AuthenticationMissmatchExceptionSerializer

- (BOOL)acceptsNull
{
	return NO;
}

- (void)write:(Kryo *)kryo value:(id)value to:(KryoOutput *)output
{
	AuthenticationMissmatchException *exception = value;
	[output writeBoolean:exception.authenticated];
}

- (id)read:(Kryo *)kryo withClass:(Class)type from:(KryoInput *)input
{
	bool authenticated = [input readBoolean];
	return [AuthenticationMissmatchException exceptionWithAuthenticated:authenticated];
}

@end
